presto-main/src/main/java/com/facebook/presto/spiller/TemporaryStoreManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.storage.TemporaryStore;
+import com.facebook.presto.spi.storage.TemporaryStoreFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.facebook.presto.spiller.LocalTemporaryStore.TEMPORARY_STORE_PATH;
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TemporaryStoreManager
+{
+    private static final Logger log = Logger.get(TemporaryStoreManager.class);
+    private static final File TEMPORARY_STORE_CONFIGURATION = new File("etc/temporary-store.properties");
```

Comment:
I don't feel very strong about this, but I would recommend allowing multiple storages, similar as we allow multiple connectors.

Basically instead of having a single file `etc/temporary-store.properties` we can have a directory, `etc/stores`. And under that directory we can have multiple configuration files, e.g.: `local.properties`, `hdfs.properties`, ...

And then some particular module can be configured to use a specific storage:

e.g.: `storage-based-spiller-store=local` or `storage-based-spiller-store=hdfs`.

That would allow different functionality to use different stores. For example spilling could use `local` store, when Broadcast join in Presto on spark would use `hdfs` store.